colorcalculator.php

```diff
@@ -0,0 +1,74 @@
+<html>
+	<body>
+		<h1> Color Calculator </h1> <br> <br>
+			<form action=""
+
+				<p>First Color
+						<select name="color1" class="color1">
+							<option value='0'>--Select function--</option>
+							<option value='Red'> Red </option>
+							<option value='Yellow'> Yellow </option>
+							<option value='Blue'> Blue </option>
+						</select>
+				</p>
+				<br>
+				<p> mixed with </p>
+				<br>
+				<p>Second Color
+						<select name="color2" class="color2">
+							<option value='0'>--Select function--</option>
+							<option value='Red'> Red </option>
+							<option value='Yellow'> Yellow </option>
+							<option value='Blue'> Blue </option>
+						</select>
+				</p>
+				<input type= "submit"/> <br>
+
+			</form>
+	</body>
+</html>
+
+<?php
+
+//all error messages
+if (isset($_REQUEST['color1']) && isset($_REQUEST['color2']) && $_REQUEST['color1']=='0') {
+	echo "<h1>Please choose two colors.</h1>";
+} elseif (isset($_REQUEST['color1']) && isset($_REQUEST['color2']) && $_REQUEST['color2']=='0') {
+	echo "<h1>Please choose two colors.</h1>";
+} elseif (isset($_REQUEST['color1']) && isset($_REQUEST['color2']) && $_REQUEST['color2']=='0' && $_REQUEST['color1']=='0') {
+	echo "<h1>Please choose two colors.</h1>";
+// all the possible colors
+} elseif (isset($_REQUEST['color1']) && isset($_REQUEST['color2']) && $_REQUEST['color1']=='Red' && $_REQUEST['color2']=='Blue') {
+	echo "<h1>Purple!</h1>";
+	echo '<body style="background-color:#d596ff">';
```

Comment:
This is the right idea, but you'd already printed the body tag earlier on the page, and you should only have one. What if instead of this approach, you calculated all these color combinations at the top of the page (before printing any html/css) and just set variables like `$backgroundColor` and `$colorText` (which would be "Yellow" etc.

Then when you print the body, just pull in that variable to set it correctly. Does that make sense?